spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -164,9 +161,45 @@ public Scan build() {
         spark, table, readConf, schemaWithMetadataColumns(), filterExpressions);
   }
 
-  public Scan buildMergeScan() {
-    return new SparkMergeScan(
-        spark, table, readConf, ignoreResiduals,
-        schemaWithMetadataColumns(), filterExpressions);
+  public Scan buildCopyOnWriteScan() {
```

Comment:
I moved the construction of `TableScan` here. I'll do the same for regular scans in the next PR.